src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -401,19 +401,22 @@ private async Task CreateExecutablesAsync(IEnumerable<AppResource> executableRes
                 var config = new Dictionary<string, string>();
                 var context = new EnvironmentCallbackContext("dcp", config);
 
+                var hasLaunchProfile = false;
                 // Need to apply configuration settings manually; see PrepareExecutables() for details.
                 if (er.ModelResource is ProjectResource project && project.SelectLaunchProfileName() is { } launchProfileName && project.GetLaunchSettings() is { } launchSettings)
                 {
                     ApplyLaunchProfile(er, config, launchProfileName, launchSettings);
+                    hasLaunchProfile = true;
                 }
-                else
+
+                // We want to make sure that certain environment variables are NOT inherited
+                foreach (var envVar in s_doNotInheritEnvironmentVars)
```

Comment:
Actually, should this just move to after `if (er.ServicesProduced.Count > 0) {}`? That way if no one sets the forbidden env variables we'll set empty ones.